src/runtime/virtcontainers/pkg/types/pcipath.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Red Hat.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package types
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+const (
+	pciSlotBits = 5
+	maxPciSlot  = (1 << pciSlotBits) - 1
+)
+
+// A PciSlot describes where a PCI device sits on a single bus
+//
+// This encapsulates the PCI slot number a.k.a device number, which is
+// limited to a 5 bit value [0x00..0x1f] by the PCI specification
+//
+// XXX In order to support multifunction device's we'll need to extend
+// this to include the PCI 3-bit function number as well.
+type PciSlot struct{ slot uint8 }
+
+func PciSlotFromString(s string) (PciSlot, error) {
+	v, err := strconv.ParseUint(s, 16, pciSlotBits)
+	if err != nil {
+		return PciSlot{}, err
+	}
+	// The 5 bit width passed to ParseUint ensures the value is <=
+	// maxPciSlot
+	return PciSlot{slot: uint8(v)}, nil
+}
+
+func PciSlotFromInt(v int) (PciSlot, error) {
+	if v < 0 || v > maxPciSlot {
+		return PciSlot{}, fmt.Errorf("PCI slot value 0x%x out of range", v)
```

Comment:
Again, reference `maxPciSlot` in error message to make it clearer.

Reply:
Good idea, done.